xarray/tests/test_formatting.py

```diff
@@ -575,17 +575,28 @@ def test_large_array_repr_length() -> None:
 
 @requires_netCDF4
 def test_repr_file_collapsed(tmp_path) -> None:
-    arr = xr.DataArray(np.arange(300), dims="test")
-    arr.to_netcdf(tmp_path / "test.nc", engine="netcdf4")
+    arr_to_store = xr.DataArray(np.arange(300, dtype=np.int64), dims="test")
+    arr_to_store.to_netcdf(tmp_path / "test.nc", engine="netcdf4")
 
     with xr.open_dataarray(tmp_path / "test.nc") as arr, xr.set_options(
         display_expand_data=False
     ):
-        actual = formatting.array_repr(arr)
+        actual = arr.__repr__()
```

Comment:
`repr(arr)` I think thats the way these dunder fucs are supposed to be called :)